lib/runner.js

```diff
@@ -161,12 +169,31 @@ var prepare = () => () => {
     });
 };
 
+var sourceDirectories = args => {
+    var elmJsonPath = path.join(args.path, 'elm.json');
+    return fs.readJson(elmJsonPath)
+        .then(elmJson => elmJson['source-directories']);
+};
+
 var allSources = args => {
+    return sourceDirectories(args)
+        .then(sourceDirs => Promise.all(sourceDirs.map(sourceDir => allSourcesFromDir(sourceDir))))
+        .then(sourcesArrays => sourcesArrays.flat());
+};
```

Comment:
This is the new machinery to read all files from all source-directories.
Potential problem: ES6 `Array.flat` usage in an otherwise quite conservative JavaScript